python/lolopy/learners.py

```diff
@@ -70,6 +70,14 @@ def fit(self, X, y, weights=None):
         # Instantiate the JVM object
         learner = self._make_learner()
 
+        # Determine the number of outputs
+        if len(y.shape) == 1:
```

Comment:
Can we be sure `y` is a NumPy array and has the `.shape` attribute (instead of something that's simply array-like)? Internally, sklearn uses [`check_array`](https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_array.html), but maybe we could use something simpler like [`np.asarray`](https://numpy.org/doc/stable/reference/generated/numpy.asarray.html).

Reply:
Just saw the `np.array` call a few lines down.  Maybe we can move that ahead of the call to `.shape`?

Reply:
The `np.array` call occurs as part of `_convert_train_data`, which has to be called after this block because it relies on `self._num_outputs` being set. I'll use `np.asarray` before calling `.shape`.